src/inference/include/openvino/runtime/properties.hpp

```diff
@@ -0,0 +1,638 @@
+    }
+
+    template <typename H>
+    static inline std::string concat(const H& head) {
+        return head;
+    }
+
+public:
+    using Key<std::string>::Key;
+
+    /**
+     * @brief Constructs device priorities
+     * @tparam Args property constructor arguments types
+     * @param args property constructor arguments
+     * @return Pair of string key representation and type erased paramter.
+     */
+    template <typename... Args>
+    inline std::pair<std::string, Any> operator()(Args&&... args) const {
+        return {str(), Any{concat(std::forward<Args>(args)...)}};
+    }
+};
+
+/**
+ * @brief Device Priorities config option, with comma-separated devices listed in the desired priority
+ */
+static constexpr Priorities priorities{"MULTI_DEVICE_PRIORITIES"};
+
+/**
+ * @brief Type for key to pass set of property values to specified device
+ */
+struct Property {
+    /**
+     * @brief Constructs property
+     * @param device_name device plugin alias
+     * @param config set of property values with names
+     * @return Pair of string key representation and type erased paramter.
+     */
+    inline std::pair<std::string, Any> operator()(const std::string& device_name, const AnyMap& config) const {
+        return {device_name, config};
+    }
+
+    /**
+     * @brief Constructs property
+     * @tparam Properties Should be the pack of `std::pair<std::string, ov::Any>` types
+     * @param device_name device plugin alias
+     * @param configs Optional pack of pairs: (config parameter name, config parameter value)
+     * @return Pair of string key representation and type erased paramter.
+     */
+    template <typename... Properties>
+    inline util::EnableIfAllProperties<std::pair<std::string, Any>, Properties...> operator()(
+        const std::string& device_name,
+        Properties&&... configs) const {
+        return {device_name, AnyMap{std::pair<std::string, Any>{configs...}}};
+    }
+};
+
+/**
+ * @brief Key to pass set of property values to specified device
+ */
+static constexpr Property properties;
+
+/**
+ * @brief Read only property to get a std::string value representing a full device name.
+ */
+static constexpr Key<std::string, PropertyMutability::RO> full_name{"FULL_DEVICE_NAME"};
+
+/**
+ * @brief Read only property which defines the device architecture.
+ */
+static constexpr Key<std::string, PropertyMutability::RO> architecture{"DEVICE_ARCHITECTURE"};
+
+/**
+ * @brief Enum to define possible device types
+ */
+enum class Type {
+    INTEGRATED = 0,  //!<  Device is integrated into host system
+    DISCRETE = 1,    //!<  Device is not integrated into host system
+};
+
+/** @cond INTERNAL */
+inline std::ostream& operator<<(std::ostream& os, const Type& device_type) {
+    switch (device_type) {
+    case Type::DISCRETE:
+        return os << "discrete";
+    case Type::INTEGRATED:
+        return os << "integrated";
+    default:
+        throw ov::Exception{"Unsupported device separability type"};
```

Comment:
separability? maybe "graphics type", cc-ing @vladimir-paramuzov

Reply:
Don't like this too. going to remove